index.js

```diff
@@ -1,67 +1,61 @@
-var isPlainObj = require('is-plain-obj')
-function defaultIsMergeable(value) {
-	return Array.isArray(value) || isPlainObj(value)
-}
+const isPlainObj = require(`is-plain-obj`)
 
-function emptyTarget(val) {
-	return Array.isArray(val) ? [] : {}
-}
+const defaultIsMergeable = value => Array.isArray(value) || isPlainObj(value)
+const emptyTarget = value => Array.isArray(value) ? [] : {}
 
-function cloneUnlessOtherwiseSpecified(value, options) {
+const cloneUnlessOtherwiseSpecified = (value, options) => {
 	return (options.clone !== false && options.isMergeable(value))
 		? deepmerge(emptyTarget(value), value, options)
 		: value
 }
 
-function defaultArrayMerge(target, source, options) {
-	return target.concat(source).map(function(element) {
-		return cloneUnlessOtherwiseSpecified(element, options)
-	})
+const defaultArrayMerge = (target, source, options) => {
+	return target.concat(source)
+		.map(element => cloneUnlessOtherwiseSpecified(element, options))
 }
 
-function getMergeFunction(key, options) {
+const getMergeFunction = (key, options) => {
 	if (!options.customMerge) {
 		return deepmerge
 	}
-	var customMerge = options.customMerge(key)
-	return typeof customMerge === 'function' ? customMerge : deepmerge
+
+	const customMerge = options.customMerge(key)
+	return typeof customMerge === `function` ? customMerge : deepmerge
 }
 
-function getEnumerableOwnPropertySymbols(target) {
+const getEnumerableOwnPropertySymbols = target => {
 	return Object.getOwnPropertySymbols
-		? Object.getOwnPropertySymbols(target).filter(function(symbol) {
-			return target.propertyIsEnumerable(symbol)
-		})
+		? Object.getOwnPropertySymbols(target)
+			.filter(target.propertyIsEnumerable)
```

Comment:
Actually, the target should be passed as a context.
Sorry for the wrong assumption.
```suggestion
			.filter(target.propertyIsEnumerable.bind(target))
```
```suggestion
			.filter(symbol => target.propertyIsEnumerable(symbol))
```